### ISSUE TITLE
Admin Interface Widget That Adds Badges to a "Bucket" Is Missable by Screen Readers

### ISSUE THREAD

**Issue by Santoz2** [ManifoldScholar/manifold#2436]
Apologies for the weird title, below is the element that I am referring to:
![image](https://user-images.githubusercontent.com/43553368/68513796-cc731000-0230-11ea-837d-0dbd400d5328.png)

This element, while it can be interacted with properly, can be missable by a Screen Reader user. For example, the "Add a Subject" box is completely missed, reading only the title, and the contents of the bucket (as it is labelled in the HTML).

## Expected Behavior
All elements of a page should be read in proper succession by a Screen Reader while browsing the page.

## Current Behavior
The widget that adds badges to buckets (eg, the "Add a Subject" field in Taxonomy) are missed by Screen Readers Browsing the page.

## Steps to Reproduce
1. Navigate to Admin Interface (via Enter Admin Mode)
2. Navigate to an editable project
3. On the General page, navigate to the Taxonomy Section
4. With a screen reader that has Browse Mode enabled (NVDA Modifier + Space if using NVDA), press the down arrow to read the following contents.
5. Note how the "Add a Subject" field is not read.
6. Navigate to the "Taxonomy" heading (NVDA can navigate headings using H or Shift + H)
7. Press Tab to navigate to the next interactable element. The next element should be the "Add a Subject" field, but focus jumps to the first badge ("Pretty", in my example)

## Testing Environment
NVDA 2019.2.1
ESR Firefox 68.2.0 esr

**Comment by dananjohnson** [comment]
@Santoz2 it's not clear to me, looking at the markup, what would cause the input to be missed by NVDA. I haven't been able to reproduce this issue with VoiceOver and Safari.

That said, I've created a pull request (#2601) that addresses a couple semantic issues with the components used here. One of the changes gives a label to the input and listbox by linking them to the header with `aria-labelledby`. I'm hopeful that this might resolve the issue you reported.

**Comment by Santoz2** [comment]
4/29/2020 Verification: There have been quite a few changes to this element. I think it is now visually clear that this is a combobox, which makes it easier for people to understand how to interact with it.

That said, the issue still persists; a user using NVDA will still miss this element if using their screen reader's browse mode. I am also unsure why this is. See below, NVDA only reports the "Subjects" label, but does not inform the user that there is a combobox there.

![image](https://user-images.githubusercontent.com/43553368/80629297-817bcd80-8a07-11ea-96bb-09ba9e4fcfc6.png)

There are comboboxes present on the rest of the page that *do* show the proper readout, reading both the title, the element type, and the contents of the element. Maybe reusing this widget for the taxonomy tags could be helpful?

![image](https://user-images.githubusercontent.com/43553368/80629527-d3bcee80-8a07-11ea-9c1d-acf4578d0505.png)

There also appears to be an issue with the badges themselves now; I believe before, navigating to them with tab would read their label, but now each item just says "Remove"

**Comment by zdavis** [comment]
The standalone mode input is just a regular HTML select input. The subjects select is a custom selection input (because it does quite a few more things in different contexts than the native select can do). We had a bunch of different versions of these complex select inputs throughout the backend, and we've now standardized on this one implementation. We're still working through some accessibility issues, @dananjohnson could you please review again based on this feedback? This is very high priority, as we need to get v5 out the door and I want to be certain this picker input meets our accessibility requirements.

Reopening issue and flagging as urgent.

**Comment by dananjohnson** [comment]
@zdavis will review today.
